Framework/PythonInterface/plugins/algorithms/HB2AReduce.py

```diff
@@ -0,0 +1,274 @@
+        self.declareProperty(FileProperty(name="Vanadium", defaultValue="", action=FileAction.OptionalLoad, extensions=[".dat", ".txt"]),
+                             doc="Vanadium file, can be either the vanadium scan file or the reduced vcorr file. "
+                             "If not provided the vcorr file adjacent to the data file will be used")
+        self.declareProperty('Normalise', True, "If False vanadium normalisation will not be performed")
+        self.declareProperty(IntArrayProperty("ExcludeDetectors", []),
+                             doc="Detectors to exclude. If not provided the HB2A_exp???__exclude_detectors.txt adjacent "
+                             "to the data file will be used if it exist")
+        self.declareProperty('IndividualDetectors', False,
+                             "If True the workspace will include each anode as a separate spectrum, useful for debugging issues")
+        condition = EnabledWhenProperty("IndividualDetectors", PropertyCriterion.IsDefault)
+        self.declareProperty('BinData', True, "Data will be binned using BinWidth. If False then all data will be unbinned")
+        self.setPropertySettings("BinData", condition)
+        positiveFloat = FloatBoundedValidator(lower=0., exclusive=True)
+        self.declareProperty('BinWidth', 0.05, positiveFloat, "Bin size of the output workspace")
+        self.setPropertySettings("BinWidth", condition)
+        self.declareProperty('Scale', 1.0, positiveFloat, "The output will be scaled by this value")
+        self.declareProperty(WorkspaceProperty("OutputWorkspace", "",
+                                               optional=PropertyMode.Mandatory,
+                                               direction=Direction.Output),
+                             "Output Workspace")
+
+    def validateInputs(self):
+        issues = dict()
+
+        if not self.getProperty("Filename").value:
+            if ((self.getProperty("IPTS").value == Property.EMPTY_INT) or
+                (self.getProperty("Exp").value == Property.EMPTY_INT) or
+               len(self.getProperty("ScanNumbers").value) is 0):
+                issues["Filename"] = 'Must specify either Filename or IPTS AND Exp AND ScanNumbers'
+
+        return issues
+
+    def PyExec(self):
+        scale = self.getProperty("Scale").value
+        filenames = self.getProperty("Filename").value
+
+        if not filenames:
+            ipts = self.getProperty("IPTS").value
+            exp = self.getProperty("Exp").value
+            filenames = ['/HFIR/HB2A/IPTS-{0}/exp{1}/Datafiles/HB2A_exp{1:04}_scan{2:04}.dat'.format(ipts, exp, scan)
+                         for scan in self.getProperty("ScanNumbers").value]
+
+        # Read in data array and append all files
+        for n, filename in enumerate(filenames):
+            # Read in all lines once
+            with open(filename) as f:
+                lines = f.readlines()
+
+            if n==0:
```

Comment:
Another way to do this is to create `metadata = None` outside the loop and change this to a check for `if metadata is None`.

Also, is metadata from summed files ignored? I would think that the user would like them appended.

Reply:
The time series properties are appended, they are part of `data` not `metadata`. I don't think it makes sense to append all the other string metadata together as most shouldn't change in the same experiment. Maybe just a limited subset, `scan` (scan number), `scan_title`, `command`, _etc_. I will think about that for a later PR.